src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/PackageFeeds/RecommenderPackageFeed.cs

```diff
@@ -0,0 +1,200 @@
+        }
+
+        public async Task<SearchResult<IPackageSearchMetadata>> RecommendPackagesAsync(ContinuationToken continuationToken, CancellationToken cancellationToken)
+        {
+            var searchToken = continuationToken as RecommendSearchToken;
+            if (searchToken == null)
+            {
+                throw new InvalidOperationException("Invalid token");
+            }
+            // don't make recommendations if the user entered a search string
+            if (!string.IsNullOrEmpty(searchToken.SearchString))
+            {
+                return SearchResult.Empty<IPackageSearchMetadata>();
+            }
+
+            // get recommender service and version info
+            if (NuGetRecommender == null)
+            {
+                NuGetRecommender = await _nuGetRecommender.GetValueAsync();
+                if (NuGetRecommender != null)
+                {
+                    var VersionDict = NuGetRecommender.GetVersionInfo();
+                    VersionInfo = (modelVersion: VersionDict.ContainsKey("Model") ? VersionDict["Model"] : (string)null,
+                                    vsixVersion: VersionDict.ContainsKey("Vsix") ? VersionDict["Vsix"] : (string)null);
+                }
+            }
+
+            List<string> recommendIds = new List<string>();
+            if (NuGetRecommender != null)
+            {
+                // get lists of only the package ids to send to the recommender
+                List<string> topPackages = _installedPackages.Select(item => item.Id).ToList();
+                // set the dependent packages to an empty list for now. We'll need to update this to the actual dependent packages
+                // when we implement PR-style projects.
+                List<string> depPackages = new List<string>();
+                // call the recommender to get package recommendations
+                recommendIds = await NuGetRecommender.GetRecommendedPackageIdsAsync(_targetFrameworks, topPackages, depPackages, cancellationToken);
+            }
+
+            if (recommendIds == null || !recommendIds.Any())
+            {
+                return SearchResult.Empty<IPackageSearchMetadata>();
+            }
+
+            // get PackageIdentity info for the top 5 recommended packages
+            int index = 0;
+            List<PackageIdentity> recommendPackages = new List<PackageIdentity>();
+            while (index < recommendIds.Count() && recommendPackages.Count < MaxRecommended)
+            {
+                MetadataResource _metadataResource = await _sourceRepository.GetResourceAsync<MetadataResource>(cancellationToken);
+                PackageMetadataResource _packageMetadataResource = await _sourceRepository.GetResourceAsync<PackageMetadataResource>(cancellationToken);
+
+                Versioning.NuGetVersion ver = await _metadataResource.GetLatestVersion(recommendIds[index], false, false, NullSourceCacheContext.Instance, Common.NullLogger.Instance, cancellationToken);
```

Comment:
nitpick: use named parameters for the two `false` parameters.

Reply:
Fixed